### ISSUE TITLE
`type+format` errors issued for `oneOf` elements in inline property schema objects

### ISSUE THREAD

**Issue by ricellis** [IBM/openapi-validator#14]
When using an identical `Schema Object` by `$ref` or inline for a property the inline case produces `type+format` warnings, but the `$ref` case (correctly I think) does not.

See attached [testoneof.yaml.zip](https://github.com/IBM/openapi-validator/files/2755463/testoneof.yaml.zip)

It contains a `/ref` path and an `/inline` path - each of which uses the same body schema of an object containing a `test` property. The `test` property's schema is defined either inline or via a `$ref`, but is otherwise identical using `oneOf`.

Unzip and run
`lint-openapi testoneof.yaml`

```
errors

  Message :   Property type+format is not well-defined.
  Path    :   paths./inline.post.requestBody.content.application/json.schema.properties.test.type
  Line    :   57
```

Even though the `/ref` path case uses an identical schema object (by reference) it does not produce the error observed in the `/inline` path case.
